src/ucp/dt/dt_generic.c

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "dt_generic.h"
+
+#include <ucs/debug/memtrack.h>
+
+
+ucs_status_t ucp_dt_create_generic(ucp_generic_dt_ops_t *ops, void *context,
+                                   ucp_datatype_t *datatype_p)
+{
+    ucp_dt_generic_t *dt;
+
+    dt = ucs_memalign(UCS_BIT(UCP_DATATYPE_SHIFT), sizeof(*dt), "generic_dt");
+    if (dt == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    dt->ops      = ops;
+    dt->context  = context;
+    *datatype_p = ((uintptr_t)dt) | UCP_DATATYPE_GENERIC;
```

Comment:
actually we have full control here because we use memalign() to get the pointer; it's not some pointer provided by the user.